Core/Utilities/ThreadPool.py

```diff
@@ -310,9 +314,22 @@ def getGlobalThreadPool():
     gThreadPool.daemonize()
   return gThreadPool
 
+gThreadPoolExecutor = False
+def getGlobalThreadPoolExecutor():
+  global gThreadPoolExecutor
+  if not gThreadPoolExecutor:
+    gThreadPoolExecutor = ThreadPoolExecutor( 500 )
+  return gThreadPoolExecutor
```

Comment:
Still, I don't get in which situation you would prefer a single ThreadPool. I see BTW this is as of now only used in the MessageBroker module. 

And, BTW, you are introducing a bug here, because in the MessageBroker modue the given ThreadPool is calling `generateJobAndQueueIt` which is of course not for `ThreadPoolExecutor`.